UndertaleModTool/Scripts/Community Scripts/TouchControls_data/gml_Object_obj_mobilecontrols_Step_0.gml

```diff
@@ -0,0 +1,146 @@
+if keyboard_check(ord("~"))
+{
+    if (device_mouse_x_to_gui(0) >= analog_posx && device_mouse_x_to_gui(0) <= (analog_posx + (59 * analog_scale)))
+        analog_center_x = (device_mouse_x_to_gui(0) - (21 * analog_scale))
+    if (device_mouse_y_to_gui(0) >= analog_posy && device_mouse_y_to_gui(0) <= (analog_posy + (59 * analog_scale)))
+        analog_center_y = (device_mouse_y_to_gui(0) - (21 * analog_scale))
+}
+else
+{
+    analog_center_x = (analog_posx + (((59 * analog_scale) / 2) - ((41 * analog_scale) / 2)))
+    analog_center_y = (analog_posy + (((59 * analog_scale) / 2) - ((41 * analog_scale) / 2)))
+}
+if keyboard_check_pressed(92)
+{
+    edit += 1
+    if (edit == 1)
+    {
+        black_fade = 0.4
+        text_black_fade = 0.9
+    }
+    else if (edit == 3)
+    {
+        virtual_key_delete(virtual_key_up)
+        virtual_key_delete(virtual_key_down)
+        virtual_key_delete(virtual_key_left)
+        virtual_key_delete(virtual_key_right)
+        virtual_key_delete(virtual_key_z)
+        virtual_key_delete(virtual_key_x)
+        virtual_key_delete(virtual_key_c)
+        virtual_key_delete(virtual_key_zp)
+        virtual_key_delete(virtual_key_xp)
+        virtual_key_delete(virtual_key_cp)
+        virtual_key_delete(virtual_key_analog)
+        virtual_key_delete(virtual_key_analogp)
+        ini_open("touchconfig.ini")
+        ini_write_real("CONFIG", "zx", zx)
+        ini_write_real("CONFIG", "zy", zy)
+        ini_write_real("CONFIG", "xx", xx)
+        ini_write_real("CONFIG", "xy", xy)
+        ini_write_real("CONFIG", "cx", cx)
+        ini_write_real("CONFIG", "cy", cy)
+        ini_write_real("CONFIG", "analog_posx", analog_posx)
+        ini_write_real("CONFIG", "analog_posy", analog_posy)
+        ini_write_real("CONFIG", "button_scale", button_scale)
+        ini_write_real("CONFIG", "analog_scale", analog_scale)
+        ini_write_real("CONFIG", "joystick_type", joystick_type)
+        ini_write_real("CONFIG", "controls_opacity", controls_opacity)
+        ini_close()
+        black_fade = 0
+        text_black_fade = 0
+        edit = 0
+        scr_add_keys()
+    }
+}
+if edit
```

Comment:
`edit` is not a bool, this seems dangerous.
Also, early return please.

Reply:
I don't remember why I wrote it like this, I'll see it in a bit.

Reply:
fixed.